client/src/featureform/register.py

```diff
@@ -265,11 +218,74 @@ def register_file(
             source (ColumnSourceRegistrar): source
         """
         FilePrefix.validate(self.__provider.config.store_type, file_path)
+        _, file_ext = os.path.splitext(file_path)
```

Comment:
use os.path.isdir?

Reply:
This only works on the current file system. I found that out the hard way 😢

Reply:
Leave a comment to save someone making the same mistake

Reply:
Good eye. The linter must not catch this. Turned on whitespace rendering and the there was indeed a mix of tabs and spaces.